src/routes/metrics.rs

```diff
@@ -57,6 +58,81 @@ pub fn query_metric_route(
     Ok(Json(results))
 }
 
+#[get("/search_parameters?<metric_name>")]
+pub fn query_metric_params(metric_name: &RawStr) -> Result<Json<MetricDataParams>, BadRequest<String>> {
+    let db_conn = establish_connection();
+    let result = metric_name.url_decode();
+    let parsed_metric_name: String;
+
+    match result {
+        Ok(o) => {
+            parsed_metric_name = o;
+        },
+        Err(_) => {
+            return Err(BadRequest(Some("bad metric_name...".to_string())));
+        }
+    }
+
+    let query_string = format!("SELECT * FROM metrics WHERE metric_name = '{}' ORDER BY id DESC LIMIT 1", parsed_metric_name);
+    println!("### query_string: {}", query_string);
+    let query_result = sql_query(query_string)
+        .load::<Metric>(&db_conn)
+        .expect("Error loading metrics");
+
+    if query_result.len() == 0 {
+        return Err(BadRequest(Some("metric name does not exist...".to_string())));
```

Comment:
Maybe note that this should be a 404 later

Reply:
yea, I didn't wanna think about it